src/tool/cpp/cppwinrt/strings/base_std_async_operation.h

```diff
@@ -4,52 +4,18 @@ WINRT_EXPORT namespace std::experimental
     template <typename TResult, typename... Args>
     struct coroutine_traits<winrt::Windows::Foundation::IAsyncOperation<TResult>, Args...>
     {
-        struct promise_type final : winrt::impl::promise_base<promise_type, winrt::Windows::Foundation::IAsyncOperation<TResult>,
+        struct promise_type final : winrt::impl::promise_base<promise_type,
+            winrt::Windows::Foundation::IAsyncOperation<TResult>,
             winrt::Windows::Foundation::AsyncOperationCompletedHandler<TResult>>
         {
-            using AsyncStatus = winrt::Windows::Foundation::AsyncStatus;
-
-            TResult GetResults()
+            TResult get_return_value() const noexcept
             {
-                winrt::slim_lock_guard const guard(this->m_lock);
-
-                if (this->m_status == AsyncStatus::Completed)
-                {
-                    return m_result;
-                }
-
-                this->rethrow_if_failed();
-                WINRT_ASSERT(this->m_status == AsyncStatus::Started);
-                throw winrt::hresult_illegal_method_call();
+                return m_result;
```

Comment:
Making this `return std::move(m_result);` might help to avoid extra copy.

Reply:
Good idea. Will do.

Reply:
I'm also going to change `return_value` to use `TResult&&` and `std::forward` to avoid a copy on the way in.